server/src/main/java/org/elasticsearch/action/support/local/LocalClusterStateRequest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.action.support.local;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.support.TransportAction;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.TimeValue;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * A base request for actions that are executed locally on the node that receives the request.
+ */
+public abstract class LocalClusterStateRequest extends ActionRequest {
+
+    private final TimeValue clusterUpdateTimeout;
```

Comment:
I was on the fence about naming this `clusterUpdateTimeout` or `masterTimeout`. Opinions are welcome :)

Reply:
Let's call it `masterTimeout` to align with the REST query parameter. If you're uncomfortable with the name, maybe add Javadoc describing what it means and how it differs from `MasterNodeRequest#masterTimeout`.